src/material/builder.rs

```diff
@@ -0,0 +1,203 @@
+use Material;
+use Texture;
+
+#[derive(Debug)]
+pub struct MeshBasic {
```

Comment:
if we are introducing the structs, then build pattern is not really necessary, strictly speaking, i.e. a material struct can implement `Default` as well as `Into<MaterialEnum>` trait returning the same enumeration type that we can serialize, and the user then does something like:
```rust
let material = three::MeshBasic { color, .. default() }.into();
```

I haven't made my mind yet on which approach is better, open for opinions!

Reply:
I'm inclined to agree.